docs/spikes/distributed_tracing.md

```diff
@@ -0,0 +1,74 @@
+- Instrument IoT Hub module with OpenTelemetry to report tracing. Since all communication on the device is routed through IoT Hub module this will give a lot of tracing data even without instrumenting other modules on the device. It will also help with use-cases like c2d, upload files, etc.
+- The key component to route OpenTelemetry tracing data to the observability backend (e.g. App Insights, Jaeger, Zipkin, etc.) is [OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md) which may work on the device as a module and in the cloud as an Azure Function or K8s microservice.
+- All modules on the device should export tracing data to OpenTelemetry Collector Module via [OpenTelemetry Protocol Exporter](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md) (e.g. with [OTLP exporter](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md)). This decouples the module code from the details on how/where the tracing data is going to be used.
+- On the devices that are mostly/normally online, the OpenTelemetry Collector Module is configured to receive traces with [OTLP Receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md) and to export traces to Azure Monitor (App Insights) via [Azure Monitor Exporter for OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter)
+  - Alternatively, the devices that are always online (can't work otherwise) and have a stable connection with the cloud (like tin LoRaWan case) may have custom modules export tracing data directly to App Insights with [Azure Monitor OpenTelemetry exporter](https://docs.microsoft.com/en-us/azure/azure-monitor/app/opentelemetry-enable?tabs=net#enable-azure-monitor-application-insights). In this case they don't need to have OpenTelemetry Collector Module on the device.
+
+![online-trace-data.png](../images/online-trace-data.png)
+
+- On the devices that may be normally offline, the OpenTelemetry Collector Module is configured to export traces to Azure Blob Storage module (an OpenTelemetry Collector exporter for that should be implemented [in this repo](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter)). Once the module is online the traces will be replicated automatically to the storage account in the cloud. On the cloud side there is an OpenTelemetry Collector instance running and receiving traces from the storage with Azure Storage receiver (to be implemented [in this repo](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver)) and exports traces to Azure Monitor via [Azure Monitor Exporter for OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter).
```

Comment:
That would be great.
- OpenTelemetry Collector is just [an engine](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/design.md) that operates receivers and exporters to pump tracing data from point A to point B. It's not designed to work in offline mode. However, I will look if it's feasible to enhance [Azure Monitor Exporter for OpenTelemetry Collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/azuremonitorexporter) so it can handle the "offline" mode. It may use the same Blob module as a temp storage.
- Another option would be to transfer trace data through the standard d2c channel which is normally used for telemetry data. This channel supports the offline mode out-of-the-box. But it would flood the channel with the trace data and it will still require additional receiver/exporter.